backend/functions/src/index.ts

```diff
@@ -18,14 +26,20 @@ const app = express()
 app.use(express.json())
 app.use(cors())
 
-// import routers
-import studentsRouter from './student/routes'
-import matchingRouter from './matching/routes'
-import courseRouter from './course/routes'
-
 // auth middleware before router
 app.use('/student', studentsRouter)
-app.use('/matching', checkAuth, matchingRouter)
-app.use('/course', checkAuth, courseRouter)
+app.use('/matching', [checkAuth, checkIsAuthorized], matchingRouter)
+app.use('/course', [checkAuth, checkIsAuthorized], courseRouter)
+
+app.get('/getauth/:email', checkAuth, (req, res) => {
+  const allowedUsersRef = db.collection('allowed_users')
+  allowedUsersRef.get().then((r) => {
+    const email = req.params.email
+    const data = r.docs.map((doc) => doc.data().email)
+    res
+      .status(200)
+      .send({ success: true, data: { isAuthed: data.includes(email) } })
+  })
+})
```

Comment:
A way you could make this more secure would be to use the request token (which is already included by the time this endpoint is accessed, plus `checkAuth` is there already) to get the email like you do with `checkIsAuthorizedHelper`, because currently this allows any signed in user to query if someone else is an administrator or not, instead of limiting it to self-checks